store/sqlstore/post_store.go

```diff
@@ -2620,64 +2620,78 @@ func (s *SqlPostStore) deleteThread(transaction *sqlxTxWrapper, postId string, d
+			sq.Eq{"Posts.UserId": userId},
+			sq.Eq{"Posts.DeleteAt": 0},
+		}).
+		ToSql()
 
-		var count int64
-		err = transaction.Get(&count, queryString, args...)
+	if err != nil {
+		return errors.Wrap(err, "failed to create SQL query to count user's posts")
+	}
 
-		if err != nil {
-			return errors.Wrap(err, "failed to count user's posts in thread")
-		}
+	var count int64
+	err = transaction.Get(&count, queryString, args...)
 
-		// Updating replyCount, and reducing participants if this was the last post in the thread for the user
-		updateQuery := s.getQueryBuilder().Update("Threads")
+	if err != nil {
+		return errors.Wrap(err, "failed to count user's posts in thread")
+	}
 
-		if count == 0 {
-			if s.DriverName() == model.DatabaseDriverPostgres {
-				updateQuery = updateQuery.Set("Participants", sq.Expr("Participants - ?", userId))
-			} else {
-				updateQuery = updateQuery.
-					Set("Participants", sq.Expr(
-						`IFNULL(JSON_REMOVE(Participants, JSON_UNQUOTE(JSON_SEARCH(Participants, 'one', ?))), Participants)`, userId,
-					))
-			}
+	// Updating replyCount, and reducing participants if this was the last post in the thread for the user
+	updateQuery := s.getQueryBuilder().Update("Threads")
+
+	if count == 0 {
+		if s.DriverName() == model.DatabaseDriverPostgres {
+			updateQuery = updateQuery.Set("Participants", sq.Expr("Participants - ?", userId))
+		} else {
+			updateQuery = updateQuery.
+				Set("Participants", sq.Expr(
+					`IFNULL(JSON_REMOVE(Participants, JSON_UNQUOTE(JSON_SEARCH(Participants, 'one', ?))), Participants)`, userId,
+				))
 		}
+	}
 
-		updateQueryString, updateArgs, err := updateQuery.
-			Set("ReplyCount", sq.Expr("ReplyCount - 1")).
-			Where(sq.And{
-				sq.Eq{"PostId": rootId},
-				sq.Gt{"ReplyCount": 0},
-			}).
-			ToSql()
+	updateQueryString, updateArgs, err := updateQuery.
+		Set("ReplyCount", sq.Expr("ReplyCount - 1")).
+		Where(sq.And{
+			sq.Eq{"PostId": rootId},
+			sq.Gt{"ReplyCount": 0},
+		}).
+		ToSql()
 
-		if err != nil {
-			return errors.Wrap(err, "failed to create SQL query to update thread")
-		}
+	if err != nil {
+		return errors.Wrap(err, "failed to create SQL query to update thread")
+	}
 
-		_, err = transaction.Exec(updateQueryString, updateArgs...)
+	if _, err = transaction.Exec(updateQueryString, updateArgs...); err != nil {
+		return errors.Wrap(err, "failed to update Threads")
+	}
+
+	updateMembershipsQueryString, args, err := s.getQueryBuilder().Update("ThreadMemberships").
+		Set("SeenReplyCount", sq.Expr("SeenReplyCount - 1")).
+		Where(sq.And{
+			sq.Eq{"PostId": rootId},
+			sq.Gt{"SeenReplyCount": 0},
+		}).ToSql()
+	if err != nil {
+		return errors.Wrap(err, "failed to create SQL query to update thread memberships")
 
-		if err != nil {
-			return errors.Wrap(err, "failed to update Threads")
-		}
 	}
-	return nil
+
+	_, err = transaction.Exec(updateMembershipsQueryString, args...)
+	return errors.Wrap(err, "failed to update ThreadMemberships")
```

Comment:
On Post delete, we don't update the `MsgCount`s in the channels, so we don't have to update channelMemberships. 

We update the ReplyCount for Threads (to show the correct reply count in the UI), so we have to update ThreadMemberships as well.